.travis.yml

```diff
@@ -3,19 +3,16 @@ language: go
 go:
   - 1.5.4
   - 1.6.3
+  - 1.7
 
 go_import_path: google.golang.org/grpc
 
 before_install:
-  - go get golang.org/x/tools/cmd/goimports
-  - go get github.com/golang/lint/golint
-  - go get github.com/axw/gocov/gocov
-  - go get github.com/mattn/goveralls
-  - go get golang.org/x/tools/cmd/cover
+  - go get -u golang.org/x/tools/cmd/goimports github.com/golang/lint/golint github.com/axw/gocov/gocov github.com/mattn/goveralls golang.org/x/tools/cmd/cover
 
 script:
-  - '! gofmt -s -d -l . 2>&1 | read'
-  - '! goimports -l . | read'
-  - '! golint ./... | grep -vE "(_string|\.pb)\.go:"'
-  - '! go tool vet -all . 2>&1 | grep -vE "constant [0-9]+ not a string in call to Errorf"'
+  - 'if [ "$TRAVIS_GO_VERSION" == 1.7 ]; then ! gofmt -s -d -l . 2>&1 | read; fi'
+  - 'if [ "$TRAVIS_GO_VERSION" == 1.7 ]; then ! goimports -l . | read; fi'
+  - 'if [ "$TRAVIS_GO_VERSION" == 1.7 ]; then ! golint ./... | grep -vE "(_string|\.pb)\.go:"; fi'
+  - 'if [ "$TRAVIS_GO_VERSION" == 1.7 ]; then ! go tool vet -all . 2>&1 | grep -vF .pb.go:; fi' # https://github.com/golang/protobuf/issues/214
```

Comment:
Because `go vet` prior to 1.7 has the `Errorf` false positive that was
removed in this change.

On Thu, Aug 18, 2016 at 1:04 PM, Chris Broadfoot <notifications@github.com>
wrote:

> In .travis.yml
> <https://github.com/grpc/grpc-go/pull/840#discussion_r75347380>:
>
> >
> >  script:
> > -  - '! gofmt -s -d -l . 2>&1 | read'
> > -  - '! goimports -l . | read'
> > -  - '! golint ./... | grep -vE "(_string|\.pb)\.go:"'
> > -  - '! go tool vet -all . 2>&1 | grep -vE "constant [0-9]+ not a string in call to Errorf"'
> > +  - 'if [ "$TRAVIS_GO_VERSION" == 1.7 ]; then ! gofmt -s -d -l . 2>&1 | read; fi'
> > +  - 'if [ "$TRAVIS_GO_VERSION" == 1.7 ]; then ! goimports -l . | read; fi'
> > +  - 'if [ "$TRAVIS_GO_VERSION" == 1.7 ]; then ! golint ./... | grep -vE "(_string|\.pb)\.go:"; fi'
> > +  - 'if [ "$TRAVIS_GO_VERSION" == 1.7 ]; then ! go tool vet -all . 2>&1 | grep -vF .pb.go:; fi' # https://github.com/golang/protobuf/issues/214
>
> Why only for 1.7?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/grpc/grpc-go/pull/840/files/61ab86bcb3b2b5987b58c9662f8976985ce4a212#r75347380>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABdsPCFui2-6itExDzNBE0Xy5W0wWdN7ks5qhJCegaJpZM4JnrCF>
> .
>